src/modules/FisheyeGl/Module.js

```diff
@@ -11,53 +11,77 @@ module.exports = function DoNothing(options, UI) {
-
-      distorter = FisheyeGl({
-        selector: '#image-sequencer-canvas'
-      });
-
-      // Parse the inputs
-      options.a = parseFloat(options.a) || distorter.lens.a;
-      options.b = parseFloat(options.b) || distorter.lens.b;
-      options.Fx = parseFloat(options.Fx) || distorter.lens.Fx;
-      options.Fy = parseFloat(options.Fy) || distorter.lens.Fy;
-      options.scale = parseFloat(options.scale) || distorter.lens.scale;
-      options.x = parseFloat(options.x) || distorter.fov.x;
-      options.y = parseFloat(options.y) || distorter.fov.y;
+      else {
+        // Create a canvas, if it doesn't already exist.
+        if (!document.querySelector('#image-sequencer-canvas')) {
+          var canvas = document.createElement('canvas');
+          canvas.style.display = 'none';
+          canvas.setAttribute('id', 'image-sequencer-canvas');
+          document.body.append(canvas);
+        }
+        else var canvas = document.querySelector('#image-sequencer-canvas');
 
-      // Set fisheyegl inputs
-      distorter.lens.a = options.a;
-      distorter.lens.b = options.b;
-      distorter.lens.Fx = options.Fx;
-      distorter.lens.Fy = options.Fy;
-      distorter.lens.scale = options.scale;
-      distorter.fov.x = options.x;
-      distorter.fov.y = options.y;
+        var distorter = FisheyeGl({
+          selector: '#image-sequencer-canvas'
+        });
 
-      // generate fisheyegl output
-      distorter.setImage(input.src, function() {
+        // Parse the inputs
+        options.a = parseFloat(options.a) || distorter.lens.a;
+        options.b = parseFloat(options.b) || distorter.lens.b;
+        options.Fx = parseFloat(options.Fx) || distorter.lens.Fx;
+        options.Fy = parseFloat(options.Fy) || distorter.lens.Fy;
+        options.scale = parseFloat(options.scale) || distorter.lens.scale;
+        options.x = parseFloat(options.x) || distorter.fov.x;
+        options.y = parseFloat(options.y) || distorter.fov.y;
 
-        // this output is accessible to Image Sequencer
-        step.output = { src: canvas.toDataURL(), format: input.format };
-
-        // Tell Image Sequencer and UI that step has been drawn
-        callback();
+        // Set fisheyegl inputs
+        distorter.lens.a = options.a;
+        distorter.lens.b = options.b;
+        distorter.lens.Fx = options.Fx;
+        distorter.lens.Fy = options.Fy;
+        distorter.lens.scale = options.scale;
+        distorter.fov.x = options.x;
+        distorter.fov.y = options.y;
+      }
+      
+      var canvas2 = document.createElement('canvas');
+      canvas2.width = pixels.shape[0]; //img.width();
+      canvas2.height = pixels.shape[1]; //img.height();
+      var ctx2 = canvas2.getContext('2d');
 
+      ctx2.putImageData(new ImageData(new Uint8ClampedArray(pixels.data), pixels.shape[0], pixels.shape[1]), 0, 0);
+      var url1 = canvas2.toDataURL();
+      
+      
+      distorter.setImage(url1, function() {
```

Comment:
This is called first time after extramanipulation is called for all frames of gifs. Is there any way to call this synchronously with each frame as the extramanipulation is called.

Reply:
I didn't get you. Could you please elaborate? Awesome work btw!

Reply:
Thanks
The only problem is that I want setImage mrthod to be called each time extraManipulation is called i.e the no. of frames in the GIF. But right now the setImage method is called for the first time after the execution of extramanipulation for all the frames are done. Some kind of callback or any way of nesting extramanipulation inside setImage will solve this problem i guess

Reply:
I'll check. Wait a min.

Reply:
Ok I checked. It is supposed to fire once for *every* frame.

Reply:
I saw the AddQR module which kind of uses similar callbacks but is able to put QR on each frame. Maybe that can help. I am working on it.